vault/external_tests/raft/raft_binary/raft_test.go

```diff
@@ -47,3 +56,377 @@ func TestRaft_Configuration_Docker(t *testing.T) {
+func removeRaftNode(t *testing.T, node *docker.DockerClusterNode, client *api.Client, serverID string) {
+	t.Helper()
+	_, err := client.Logical().Write("sys/storage/raft/remove-peer", map[string]interface{}{
+		"server_id": serverID,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	node.Cleanup()
+	// TODO: remove from the cluster nodes as well.
+}
+
+// stabilizeAndPromote makes sure the given node ID is among the voters using
+// autoPilot state
+func stabilizeAndPromote(t *testing.T, client *api.Client, nodeID string) {
+	t.Helper()
+	deadline := time.Now().Add(2 * autopilot.DefaultReconcileInterval)
+	failed := true
+	var state *api.AutopilotState
+	var err error
+	for time.Now().Before(deadline) {
+		state, err = client.Sys().RaftAutopilotState()
+		require.NoError(t, err)
+		if state != nil && state.Servers != nil && state.Servers[nodeID].Status == "voter" {
+			failed = false
+			break
+		}
+		time.Sleep(1 * time.Second)
+	}
+
+	if failed {
+		t.Fatalf("autopilot failed to promote node: id: %#v: state:%# v\n", nodeID, state)
+	}
+}
+
+// stabilize makes sure the cluster is in a healthy state using autopilot state
+func stabilize(t *testing.T, client *api.Client) {
+	t.Helper()
+	deadline := time.Now().Add(2 * autopilot.DefaultReconcileInterval)
+	healthy := false
+	for time.Now().Before(deadline) {
+		state, err := client.Sys().RaftAutopilotState()
+		require.NoError(t, err)
+		if state.Healthy {
+			healthy = true
+		}
+		time.Sleep(1 * time.Second)
+	}
+	if !healthy {
+		t.Fatalf("cluster failed to stabilize")
+	}
+}
+
+// TestDocker_LogStore_Boltdb_To_Raftwal_And_Back runs 3 node cluster leveraging boltDB
+// as the logStore, then migrates the cluster to raft-wal logStore and back.
+// This shows raft-wal does not lose data.
+// There is no migration procedure for individual nodes.
+// The correct procedure is destroying existing raft-boltdb nodes and starting brand-new
+// nodes that use raft-wal (and vice-versa)
+// Having a cluster of mixed nodes, some using raft-boltdb and some using raft-wal, is not a problem.
+func TestDocker_LogStore_Boltdb_To_Raftwal_And_Back(t *testing.T) {
+	t.Parallel()
+	binary := os.Getenv("VAULT_BINARY")
+	if binary == "" {
+		t.Skip("only running docker test when $VAULT_BINARY present")
+	}
+	opts := &docker.DockerClusterOptions{
+		ImageRepo: "hashicorp/vault",
+		// We're replacing the binary anyway, so we're not too particular about
+		// the docker image version tag.
+		ImageTag:    "latest",
+		VaultBinary: binary,
+		ClusterOptions: testcluster.ClusterOptions{
+			VaultNodeConfig: &testcluster.VaultNodeConfig{
+				LogLevel: "TRACE",
+			},
+		},
+	}
+	cluster := docker.NewTestDockerCluster(t, opts)
+	defer cluster.Cleanup()
+
+	rafttest.Raft_Configuration_Test(t, cluster)
+
+	leaderNode := cluster.GetActiveClusterNode()
+	leaderClient := leaderNode.APIClient()
+
+	err := leaderClient.Sys().MountWithContext(context.TODO(), "kv", &api.MountInput{
+		Type:  "kv",
+		Local: true,
```

Comment:
Do we want this to be a local mount? I confess this is an area I don't know a ton about 🤔 Does this mean it's just local for the purposes of DR/PR replication? If so, then it probably doesn't matter, since we're not bothering with any of that here.